.github/workflows/build.yml

```diff
@@ -50,10 +50,74 @@ jobs:
       - name: "Install Python dependencies for PDF Generator"
         run: pip install -r service-python/pdfgenerator/src/requirements.txt
 
+      - name: Restore cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle
+          restore-keys: |
+            v1-dependencies-${{ hashFiles('build.gradle') }}
+            v1-dependencies-
+          key: v1-dependencies-${{ hashFiles('build.gradle') }}
+
+      - name: Setup gradle
+        run: ./gradlew dependencies
+
       # Build and test
 
       - name: Build and test Java code
-        run: ./gradlew build check
+        run: ./gradlew build check -PGITHUB_ACCESS_TOKEN=${{ env.GITHUB_ACCESS_TOKEN }}
 
       - name: "Build and test Python code with test coverage report"
         run: cd service-python && python -m pytest --cov-report term-missing --cov
+
+  build-docker-app:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          install: true
+      - name: Build
+        run: |
+          echo "GITHUB_ACCESS_TOKEN=${{ secrets.LHDI_PAT }}" > all-secrets.txt
+          docker build . --no-cache --progress=plain --secret id=all-secrets,src=all-secrets.txt
+          rm all-secrets.txt
+        working-directory: ./code
+
+  build-docker-db:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          install: true
+
+      - name: Build
+        run: |
+          docker build -f ./db-init/src/docker/Dockerfile .
+        working-directory: ./code
+
+  validate-new-app-generation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: '17'
+
+      - name: Install cookiecutter
+        run: pip install cookiecutter==1.7.3
+
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Compare changes in template
+        run: ./gradlew ccdiff -PGITHUB_ACCESS_TOKEN=${{ env.GITHUB_ACCESS_TOKEN }} --info --stacktrace
```

Comment:
When this change was made in LH StarterKit:
https://github.com/department-of-veterans-affairs/lighthouse-di-demo-java-starterkit/commit/eb6d84fd3cdc02ab2ac97875af086598b53514d3?diff=split#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3